internal/acctest/backward_compat.go

```diff
@@ -84,6 +84,12 @@ type BackwardCompatConfig struct {
 
 	// OldProviderVersion is the version to test against (defaults to latest stable)
 	OldProviderVersion string
+
+	// PlanOnly controls whether verification step runs with PlanOnly mode (defaults to false).
+	// When false, verification step performs a refresh during plan (`terraform plan`) to allow Read functions to migrate state.
+	// When true, verification only compares state to schema without calling Read functions
+	// (equivalent to `terraform plan -refresh=false`).
+	PlanOnly bool
```

Comment:
nit: the description is clear and correct. Though I suggest adding an example where it comes handy or important. This PR is the one that ran into the case where that matters: refreshing the plan looses certain information which is important for computed and wo fields (I guess).